homeassistant/components/incomfort/config_flow.py

```diff
@@ -102,6 +108,43 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return InComfortOptionsFlowHandler()
 
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a DHCP discovered Intergas Gateway device."""
+        self._discovered_host = discovery_info.ip
+        # In case we have an existing entry with the same host
+        # we update the entry with the unique_id for the gateway, and abort the flow
+        unique_id = format_mac(discovery_info.macaddress)
+        existing_entries = [
+            entry
+            for entry in self._async_current_entries(include_ignore=False)
+            if entry.unique_id is None
+            and entry.data.get(CONF_HOST) == self._discovered_host
+        ]
+        if existing_entries:
+            self.hass.config_entries.async_update_entry(
+                existing_entries[0], unique_id=unique_id
+            )
+            self.hass.config_entries.async_schedule_reload(existing_entries[0].entry_id)
+            raise AbortFlow("already_configured")
+
+        await self.async_set_unique_id(unique_id)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: self._discovered_host})
+
+        return await self.async_step_dhcp_confirm()
+
+    async def async_step_dhcp_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm setup from discovery."""
+        if user_input is not None:
+            return await self.async_step_user({CONF_HOST: self._discovered_host})
+        return self.async_show_form(
+            step_id="dhcp_confirm",
+            description_placeholders={CONF_HOST: self._discovered_host},
+        )
```

Comment:
We know the host, so we only need to ask the user for username and password at this point

Reply:
True, but why not show the user the IP address/host found and reuse the code?

Reply:
I made an extra flow step `dhcp_auth` so that only the username and password are configurable in case authentication is needed